test/integration/garbage_collector_test.go

```diff
@@ -357,7 +359,7 @@ func TestStressingCascadingDeletion(t *testing.T) {
 	go gc.Run(5, stopCh)
 	defer close(stopCh)
 
-	const collections = 30
+	const collections = 10
```

Comment:
I'm a little surprised that 30 collections takes 6 minutes to test. I'm going to LGTM this to get the running time down, but maybe you can look into seeing why it is slow?